odl/tomo/data/uncompr_bin.py

```diff
@@ -0,0 +1,472 @@
+                continue
+
+            for key, val in row.items():
+                fields[-1][key.strip()] += (' ' + val).rstrip()
+
+    return tuple(fields)
+
+
+def standardized_fields(field_list, keys, dtype_map):
+    """Convert the field keys and values to standard format.
+
+    Data type is piped through `numpy.dtype`, and name is converted
+    to lowercase. All keys are converted to lowercase.
+
+    The standardized fields are as follows:
+
+    +--------------+---------+------------------------------------------+
+    |Name          |Data type|Description                               |
+    +==============+=========+==========================================+
+    |'name'        |string   |Name of the element                       |
+    +--------------+---------+------------------------------------------+
+    |'offset_bytes'|int      |Offset of the current element in bytes    |
+    +--------------+---------+------------------------------------------+
+    |'size_bytes'  |int      |Size of the current element in bytes      |
+    +--------------+---------+------------------------------------------+
+    |'dtype'       |type     |Data type of the current element as       |
+    |              |         |defined by Numpy                          |
+    +--------------+---------+------------------------------------------+
+    |'description' |string   |Description of the element (optional)     |
+    +--------------+---------+------------------------------------------+
+    |'dshape'      |tuple    |For multi-elements: number of elements per|
+    |              |         |dimension. Optional for single elements.  |
+    +--------------+---------+------------------------------------------+
+
+    Parameters
+    ----------
+    field_list : sequence of dicts
+        Dictionaries describing the field elements, as returned by, e.g.,
+        `mrc_spec_fields`.
+    keys : dict
+        Dictionary with the following entries for the column headers in
+        the specification table:
+
+            - ``'id'``
+            - ``'byte_range'``
+            - ``'dtype'``
+            - ``'name'``
+            - ``'description'``
+
+    dtype_map : dict
+        Mapping from the data type specifiers in the specification table
+        to NumPy data types.
+
+    Returns
+    -------
+    standardized_fields : tuple of dicts
+        The standardized fields according to the above table.
+    """
+    # Parse the fields and represent them in a unfied way
+    conv_list = []
+    for row, field in enumerate(field_list):
+        new_field = {}
+
+        # Name and description: lowercase name, copy description
+        new_field['name'] = field[keys['name']].lower()
+        new_field['description'] = field[keys['description']]
+
+        # Get offset from ID range
+        num_range = field[keys['id']].split('-')
+        nstart = int(num_range[0])
+        nend = int(num_range[-1])  # 0 for range of type 3-
```

Comment:
Not correct: `num_range[-1] == ''` in this case, which raises an error when passed to `int`.